test/addons/node-module-version/test.js

```diff
@@ -4,8 +4,11 @@ const common = require('../../common');
 const assert = require('assert');
 
 const re = new RegExp(
-  'was compiled against a different Node.js version using\n' +
-  'NODE_MODULE_VERSION 42. This version of Node.js requires\n' +
-  `NODE_MODULE_VERSION ${process.versions.modules}.`);
+  `^Error: The module '.+'\n` +
```

Comment:
Most node `Error: ` messages don't insert line breaks into the message output.

Consider removing the newlines, so that the message can wrap at the user's terminal boundar, of in the case of stderr being directed to a log aggregation service, the entire error will be one single line/log entry.

I would make each sentence a single line, or maybe the entire thing two lines, with the second line being "Please try...".